src/main/kotlin/Stopwatch.kt

```diff
@@ -1,37 +1,42 @@
-import java.util.concurrent.CopyOnWriteArraySet
+import io.reactivex.Observable
+import io.reactivex.Scheduler
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
+import java.util.concurrent.TimeUnit
 import java.util.concurrent.atomic.AtomicLong
 
-interface Stopwatch {
-
-    interface StopwatchListener {
-        fun onValueChange(value: Long)
-    }
 
-    fun resumeStopwatch(stopwatchListener: StopwatchListener)
-    fun pauseStopwatch(stopwatchListener: StopwatchListener)
-
-    class Impl(private val ticker: Ticker) : Stopwatch, Ticker.TickListener {
-
-        private val listeners = CopyOnWriteArraySet<StopwatchListener>()
-        private val value = AtomicLong(0)
+interface Stopwatch {
 
-        override fun onTick() {
-            val newValue = value.incrementAndGet()
-            listeners.forEach { it.onValueChange(newValue) }
+    val value: Subject<Long>
+    fun pause()
+    fun resume()
+
+    class Impl(private val scheduler: Scheduler) : Stopwatch {
+
+        override val value =  BehaviorSubject.create<Long>().toSerialized()
+        private val consumersCountChange = PublishSubject.create<Long>().toSerialized()
+        private val actualValueCounter = AtomicLong()
+
+        init {
+            consumersCountChange
+                .scan{ x, y -> x + y }
+                .filter{ x -> x in 0..1}
+                .switchMap { o -> if (o < 1) Observable.never<Long>()
+                                  else Observable
+                                                    .interval(1, TimeUnit.SECONDS, scheduler)
+                                                    .map{ _ -> actualValueCounter.getAndIncrement() } }
+                .subscribe(value)
+            value.share()
         }
 
-        override fun resumeStopwatch(stopwatchListener: StopwatchListener) {
-            listeners.add(stopwatchListener)
-            if (listeners.size == 1) {
-                ticker.start(this)
-            }
+        override fun pause() {
+            consumersCountChange.onNext(-1)
         }
 
-        override fun pauseStopwatch(stopwatchListener: StopwatchListener) {
-            listeners.remove(stopwatchListener)
-            if (listeners.size == 0) {
-                ticker.stop()
-            }
+        override fun resume() {
+            consumersCountChange.onNext(1)
         }
```

Comment:
Let’s use `+1` for clarity and symmetry with `-1`. Also, it can be written as:
```kotlin
override fun resume() = consumersCountChange.onNext(1)
```